File: implementations/node/sdk/src/contracts/executions/execution-submission-response.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionStatus } from "./execution-status.js";

export interface AiSdkExecutionSubmissionResponse {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionSubmissionResponse;
  readonly executionId: string;
  readonly publicationRef: string;
  readonly status: AiSdkExecutionStatus;
  readonly acceptedAtUtc: string;
  readonly idempotencyKey?: string;
  readonly correlationId?: string;
}

File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/executions/execution_submission_response.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from .execution_status import AiSdkExecutionStatus


@dataclass(frozen=True)
class AiSdkExecutionSubmissionResponse(AiSdkWireModel):
    execution_id: str
    publication_ref: str
    status: AiSdkExecutionStatus
    accepted_at_utc: str
    schema_version: int = AiSdkSchemaVersions.EXECUTION_SUBMISSION_RESPONSE
    idempotency_key: str | None = None
    correlation_id: str | None = None
